src/fenwick.jl

```diff
@@ -0,0 +1,96 @@
+"""
+    FenwickTree{T}(n)
+    
+Constructs a [`FenwickTree`](https://en.wikipedia.org/wiki/Fenwick_tree) of length `n`.
+ 
+"""
+mutable struct FenwickTree{T}
+    bit::Vector{T}
+    n::Integer
+end
+
+FenwickTree{T}() where T = FenwickTree{T}(0)
+FenwickTree{T}(n::Integer) where T = FenwickTree{T}(zeros(T, n), n)
+
+"""
+    FenwickTree(arr::AbstractArray) 
+    
+Constructs a [`FenwickTree`](https://en.wikipedia.org/wiki/Fenwick_tree) from `arr`.
+ 
+"""
+function FenwickTree(a::AbstractVector{U}) where U
+    n = length(a)
+    bit = FenwickTree{U}(n)
+    @inbounds for i = 1:n
+        update!(bit, i, a[i])
+    end
+    bit
+end
+
+bit(F::FenwickTree{T}) where T = F.bit
```

Comment:
How frequent is it that a user would want to access the underlying tree?  It seems that the prefix sum is the most important aspect of the tree.

Unless you expect this to be accessed quite frequently by users, I would suggest removing it.  The underlying data store can be accessed using `f.binary_index_tree` (or whatever it ends up being called).

Reply:
Also, in julia, types are usually CamelCase, and variables are lower case. So `F` should be `f` or `ft` (here and in most of the function definitions below).